src/CLI/ProcessHelper.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+
+namespace AxeWindowsScanner
+{
+    public class ProcessHelper : IProcessHelper
+    {
+        private readonly IProcessAbstraction _processAbstraction;
+        private readonly IErrorCollector _errorCollector;
+
+        public ProcessHelper(IProcessAbstraction processAbstraction, IErrorCollector errorCollector)
+        {
+            if (processAbstraction == null) throw new ArgumentNullException(nameof(processAbstraction));
+            if (errorCollector == null) throw new ArgumentNullException(nameof(errorCollector));
+            _processAbstraction = processAbstraction;
+            _errorCollector = errorCollector;
+        }
+
+        public int FindProcessByName(string processName)
+        {
+            var processes = _processAbstraction.GetProcessesByName(processName);
+
+            if (processes == null || processes.Length == 0)
+            {
+                _errorCollector.AddParameterError("Unable to find process with name " + processName);
+                return IProcessHelper.InvalidProcessId;
```

Comment:
I realize this is a little easier, but I think throwing an exception might make the code a better. Dr. Bob recommends throwing exceptions rather than returning "magic" numbers. (Feel free to push back.)

Reply:
Dr. Dave generally concurs, but this patient makes it difficult. This code gets called by the ctor for the Options class, which is called by CommandLineParser (not our code). CommandLineParser eats all Exceptions without allowing us to access the original information, so the final user experience is unpleasant. To get around that, I defined this interface to not throw (for the sake of messaging), which meant that it needed to return _something_ recognizable.